src/EffectBase.h

```diff
@@ -91,12 +100,29 @@ namespace openshot
 
 		/// Set parent clip object of this effect
 		void ParentClip(openshot::ClipBase* new_clip);
+		
+		/// Set the parent effect from which this properties will be set to
+		void SetParentEffect(std::string parentEffect_id);
+
+		/// Return the ID of this effect's parent clip
+		std::string ParentClipId() const;
+
+		/// Get the indexes and IDs of all visible objects in the given frame
+		virtual std::string GetVisibleObjects(int64_t frame_number) const {return {}; };
```

Comment:
Here too, "visible" seems like a very specific concept with no generalizable meaning that would warrant its inclusion in the base class. How would a Blur effect or a Brightness effect know what objects are visible in the current frame? What does that even _mean_, for a Blur or Brightness effect?

(Also, the documentation comment says "Get the indexes and IDs", but the method returns a `std::string`. So, feels like there's detail missing there.)